practice-app/website/api/badge.py

```diff
@@ -29,4 +29,25 @@ def add_badge():
         db.session.rollback()
         return "Try Later", 403
         
-    return jsonify(badge.serialize()), 201
\ No newline at end of file
+    return jsonify(badge.serialize()), 201
+
+@badges.route('show_badge/',methods = ['GET','POST'])
+def show_badge():
+    if request.method == 'POST':
```

Comment:
this method does not seem to be a post method. it does not insert aything
may it be a get request?

Reply:
Yes, what it does is getting some data from the database. However it works when user clicks the button which sends post request to the backend. That is why it checks whether it is a post request or not. It starts working with a post request from the user, however it is a get request in the end. It will be more clear after I write the comments.

Reply:
i think get/post methods between frontend and backend is different from get/post methods between backend and api. if you change 'response = requests.post(req, headers=headers)' to 'response = requests.get(req, headers=headers)' in views.py line 85, and also change the check from post to get, functionality will remain and methods will be more clear

Reply:
You are right. I changed it.